docs/src/reference/the-traversal.asciidoc

```diff
@@ -1434,6 +1436,15 @@ the key,value pairs for those vertices.
 <6> Find all age-properties and emit their value.
 <7> Find all vertices that do not have an age-property and emit their name.
 <8> Find all "person" vertices that have a name property that starts with the letter "m".
+<9> Property key values cannot be `null` and therefore produces no matches.
+<10> An example of `has()` where the argument is a `Traversal` and does not quite behave the way most expect.
+
+Item 10 in the above set of examples bears some discussion. The behavior is not such that the result of the `Traversal`
+is used as the comparing value for `has()`, but the current `Traverser`, which in this case is the vertex `label`, is
+given to the `Traversal` to behave as a filter itself. In other words, if the `Traversal` (i.e. `is('person')`) returns
+a value then the `has()` is effectively `true`. A common mistake is to try to use `select()` in this context where one
+would do `has('name', select('n'))` to try to inject the value of "n" into the step to get `has('name', <value-of-n>)`,
```

Comment:
Oh wow, good explanation. I expected `has()` to behave this way with the `is()` step, but I would also have expected it to work with `select()` like you described here. Good to know!